### DISCUSSION TITLE
El Select create/filter formatter or validator

### DISCUSSION THREAD

**Discussion by Mark-ICS** [element-plus/element-plus#14397] [Q&A]
I have an el-select component which I am using like this:

```
<el-select v-model="selectedItems" placeholder="Items" filterable allow-create multiple>
    <el-option v-for="item in items" :key="item.value" :label="item.label" :value="item.value" />
</el-select>
```

This allows me to create new items as well as filter and select from existing items.

My question is: Is there an equivalent to `:formatter` on the `<el-input>` tag for the `<el-select>` that is being used to both filter and create items?

My goal is to only allow users to enter lowercase a-z characters.

Basically, I want to restrict which characters can be entered into the filter?

**Comment by ryuhangyeong** [comment]
[Here Link](https://element-plus.org/en-US/component/select.html#select-attributes:~:text=false-,filter%2Dmethod,-custom%20filter%20method)

Is this the property you want?

**Reply by Mark-ICS** [reply]
Sorry I don't see which one the link is referring too?

**Reply by ryuhangyeong** [reply]
`filter-method`

**Comment by xing403** [comment]
Use the `:filter-method` attribute and `@change` event in the `el-select` component
such as:
```html
<el-select v-model="value" filterable allow-create :filter-method="filterMethod" @change="handleChangeSelect">
  <el-option v-for="item in options" :key="item.value" :label="item.label" :value="item.value" />
</el-select>
```

```ts
// custom filter method
function filterMethod(keyword: string) {
  options.value = RawData.filter((item) => {
    return item.label.toLowerCase().startsWith(keyword.toLowerCase())
  })
}
// After the user selects, the content is changed to lower case. You can also customize the format in here
function handleChangeSelect(val: string) {
  // value.value = val.replace('-', '/') // `-` Convert to `/`
  value.value = val.toLowerCase()
}
```
